### DISCUSSION TITLE
What format should the 'spacing' parameter be in the MONAI compute_hausdorff_distance function?

### DISCUSSION THREAD

**Discussion by KateElizabeth112** [Project-MONAI/MONAI#7329] [Q&A]
I am trying to use the compute_hausdorff_distance function with the spacing parameter set to the voxel spacing. It worked just fine with spacing set to None, but when I try and set it to my voxel spacing I get an error:

```
AssertionError: `spacing` is a sequence of elements with unsupported type: <class 'numpy.float32'>
```

I tried to look at the function impelementation, and it seems that float values should be fine: 
![image](https://github.com/Project-MONAI/MONAI/assets/15280385/76875047-e73b-48f0-b90a-7d28428e534a)

Here's how I get the voxel spacing in my own code: 
```
# get the volume of 1 voxel in mm3
sx, sy, sz = gt_nii.header.get_zooms()
vox_vol = sx * sy * sz
vox_spacing = np.array([sx, sy, sz])
```

I would really appreciate some guidance on how to format the spacing so it doesn't cause an error.

**Comment by KumoLiu** [comment]
Hi @KateElizabeth112, perhaps you should specify an array directly or `[sx, sy, sz]` directly.
Hope it helps, thanks!

**Reply by KateElizabeth112** [reply]
Thank you, I tried that as well, but it turns out the issue is not to do with the format (list or numpy array) but the data type of the elements. sx, sy and sz were numpy floats (which is not supported) so I changed them to Python floats in the following way and it worked: 

```
vox_spacing = [sx.item(), sy.item(), sz.item()]
```
